homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,463 @@
+        """Return the maximum temperature."""
+        return self.target_temperature_high
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return self.target_temperature_low
+
+    @property
+    def target_temperature_high(self) -> float:
+        """Return the upper bound target temperature."""
+        if self.__is_celsius():
+            if DPCODE_TEMP_SET not in self.tuya_device.function:
+                return 0
+            temp_value = json.loads(
+                self.tuya_device.function.get(DPCODE_TEMP_SET, {}).values
+            )
+            return temp_value.get("max", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+        if DPCODE_TEMP_SET_F not in self.tuya_device.function:
+            return 0
+        temp_value = json.loads(
+            self.tuya_device.function.get(DPCODE_TEMP_SET_F, {}).values
+        )
+        return temp_value.get("max", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Return the lower bound target temperature."""
+        if self.__is_celsius():
+            if DPCODE_TEMP_SET not in self.tuya_device.function:
+                return 0
+            temp_value = json.loads(
+                self.tuya_device.function.get(DPCODE_TEMP_SET, {}).values
+            )
+            low_value = (
+                temp_value.get("min", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+            )
+            return low_value
+        if DPCODE_TEMP_SET_F not in self.tuya_device.function:
+            return 0
+        temp_value = json.loads(
+            self.tuya_device.function.get(DPCODE_TEMP_SET_F, {}).values
+        )
+        return temp_value.get("min", 0) * 1.0 / (10 ** self.get_temp_set_scale())
+
+    @property
+    def target_temperature_step(self) -> float:
+        """Return target temperature setp."""
+        if (
+            DPCODE_TEMP_SET not in self.tuya_device.status_range
+            and DPCODE_TEMP_SET_F not in self.tuya_device.status_range
+        ):
+            return 1.0
+        __temp_set_value_range = json.loads(
+            self.tuya_device.status_range.get(
+                DPCODE_TEMP_SET if self.__is_celsius() else DPCODE_TEMP_SET_F
+            ).values
+        )
+        return (
+            __temp_set_value_range.get("step", 0)
+            * 1.0
+            / (10 ** self.get_temp_set_scale())
+        )
+
+    @property
+    def target_humidity(self) -> int:
+        """Return target humidity."""
+        return int(self.tuya_device.status.get(DPCODE_HUMIDITY_SET, 0))
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac mode."""
+        if not self.tuya_device.status.get(DPCODE_SWITCH, False):
+            return HVAC_MODE_OFF
+        if DPCODE_MODE not in self.tuya_device.status:
+            return HVAC_MODE_OFF
+        return TUYA_HVAC_TO_HA[self.tuya_device.status.get(DPCODE_MODE)]
```

Comment:
`None` doesn't exist in `TUYA_HVAC_TO_HA`. If `DPCODE_MODE` is missing in the status dict the lookup will return `None` and there will be a `KeyError` here.

Reply:
fixed